crew_testing/tools/openapi_auth_detector.py

```
@@ -0,0 +1,109 @@
+                operation = path_item.get(method)
+                if operation and "security" in operation:
+                    # Endpoint has security requirements
+                    auth_required = True
+                    for security_req in operation["security"]:
+                        # security_req is like {"HTTPBearer": []}
+                        auth_types.update(security_req.keys())
+
+            if auth_required:
+                return json.dumps({
+                    "requires_auth": True,
+                    "auth_type": list(auth_types)[0] if auth_types else "HTTPBearer",
```

Ruff (0.14.3)

Prefer `next(iter(auth_types))` over single element slice

Replace with `next(iter(auth_types))`

(RUF015)